core/lib/src/response/response.rs

```diff
@@ -354,15 +383,21 @@ impl<'r> ResponseBuilder<'r> {
     /// # #[allow(unused_variables)]
     /// let response = Response::build()
     ///     .sized_body(File::open("body.txt").await?)
-    ///     .finalize();
+    ///     .await;
     /// # Ok(())
     /// # }
     /// ```
     #[inline(always)]
-    pub fn sized_body<B>(&mut self, body: B) -> &mut ResponseBuilder<'r>
-        where B: AsyncRead + io::Seek + Send + Unpin + 'r
+    pub fn sized_body<B: 'r>(&mut self, mut body: B) -> &mut ResponseBuilder<'r>
+        where B: AsyncRead + AsyncSeek + Send + Unpin + 'r
     {
-        self.response.set_sized_body(body);
+        self.sized_body = Some(Box::pin(async {
```

Comment:
This duplicates the logic in `Response`. Let's try changing this to `self.pending_sized_body` as a `Box<dyn AsyncRead + AsyncSeek + Send + Unpin + 'r>`, and call `response.set_sized_body` inside `async fn finalize`. If necessary we can split out another method or two to reduce these extraneous `Box`es.